app/models/megaphone/episode.rb

```diff
@@ -191,9 +212,31 @@ def audio_is_latest?
     end
 
     def set_enclosure
-      return unless audio_file_status == "success"
-      return if download_url.blank? || feeder_episode.enclosure_override_url.present?
-      feeder_episode.update(enclosure_override_url: download_url, enclosure_override_prefix: true)
+      return if download_url.blank? || !feeder_episode.complete_media?
+      set_external_media
+      feeder_episode.update!(enclosure_override_url: download_url, enclosure_override_prefix: true)
+    end
```

Comment:
Definite changes here!
first off, don't try to set the enclosure override if no audio has been uploaded to DT or sent to megaphone

next, don't retrieve the attributes by requesting from megaphone - 
it won't work while it's a draft, and since we uploaded it, we know all the attributes

finally, if updating does fail, raise an error